include/libsemigroups/ukkonen.hpp

```diff
@@ -878,343 +1003,664 @@ namespace libsemigroups {
     void tree_extend(index_type pos);
   };
 
+  //! \brief Namespace for Ukkonen helper functions.
+  //!
+  //! This namespace contains helper functions for the Ukkonen class.
   namespace ukkonen {
-    //! Add all words in a \vector to a Ukkonen object.
+    //! \brief \copybrief Ukkonen.add_word_no_checks
+    //!
+    //! See \ref Ukkonen.add_word_no_checks.
+    inline void add_word_no_checks(Ukkonen& u, word_type const& w) {
+      u.add_word_no_checks(w.cbegin(), w.cend());
+    }
+
+    //! \brief \copybrief Ukkonen.add_word_no_checks
+    //!
+    //! See \ref Ukkonen.add_word_no_checks.
+    template <typename Iterator>
+    void add_word_no_checks(Ukkonen& u, Iterator first, Iterator last) {
+      // TODO(later) it'd be better to just convert the values pointed at by the
+      // iterators, than to allocate a word_type here, but this is currently a
+      // bit tricky to set up
+      add_word_no_checks(u, word_type(first, last));
+    }
+
+    //! \brief \copybrief Ukkonen.add_word_no_checks
+    //!
+    //! See \ref Ukkonen.add_word_no_checks.
+    template <typename Word>
+    void add_word_no_checks(Ukkonen& u, Word const& w) {
+      add_word_no_checks(u, w.cbegin(), w.cend());
+    }
+
+    //! \brief \copybrief Ukkonen.add_word_no_checks
+    //!
+    //! See \ref Ukkonen.add_word_no_checks.
+    inline void add_word_no_checks(Ukkonen& u, char const* w) {
+      add_word_no_checks(u, w, w + std::strlen(w));
+    }
+
+    //! \brief \copybrief Ukkonen.add_word_no_checks
+    //!
+    //! See \ref Ukkonen.add_word_no_checks.
+    //!
+    //! \throws LibsemigroupsException if `throw_if_not_unique_letters(w)`
+    //! throws.
+    //!
+    //! \sa \ref Ukkonen.throw_if_not_unique_letters.
+    inline void add_word(Ukkonen& u, word_type const& w) {
+      u.add_word(w.cbegin(), w.cend());
+    }
+
+    //! \brief \copybrief Ukkonen.add_word_no_checks
+    //!
+    //! See \ref Ukkonen.add_word_no_checks.
+    //!
+    //! \throws LibsemigroupsException if `throw_if_not_unique_letters(first,
+    //! last)` throws.
+    //!
+    //! \sa \ref Ukkonen.throw_if_not_unique_letters.
+    template <typename Iterator>
+    void add_word(Ukkonen& u, Iterator first, Iterator last) {
+      // TODO(later) it'd be better to just convert the values pointed at by the
+      // iterators, than to allocate a word_type here, but this is currently a
+      // bit tricky to set up
+      add_word(u, word_type(first, last));
+    }
+
+    //! \brief \copybrief Ukkonen.add_word_no_checks
+    //!
+    //! See \ref Ukkonen.add_word_no_checks.
+    //!
+    //! \throws LibsemigroupsException if `throw_if_not_unique_letters(w)`
+    //! throws.
+    //!
+    //! \sa \ref Ukkonen.throw_if_not_unique_letters.
```

Comment:
```suggestion
    //! \sa \ref Ukkonen::throw_if_not_unique_letters.
```
Some python slipped back in here